server/channels/app/post.go

```diff
@@ -2242,3 +2242,255 @@ func (a *App) GetPostInfo(c request.CTX, postID string) (*model.PostInfo, *model
 	}
 	return &info, nil
 }
+
+func makePostLink(siteURL, teamName, postID string) string {
+	return fmt.Sprintf("%s/%s/pl/%s", siteURL, teamName, postID)
+}
+
+// validateMoveOrCopy performs validation on a provided post list to determine
+// if all permissions are in place to allow the for the posts to be moved or
+// copied.
+func (a *App) ValidateMoveOrCopy(c *request.Context, wpl *model.WranglerPostList, originalChannel *model.Channel, targetChannel *model.Channel, user *model.User) error {
+	if wpl.NumPosts() == 0 {
+		return errors.New("The wrangler post list contains no posts")
+	}
+
+	config := a.Config().WranglerSettings
+
+	switch originalChannel.Type {
+	case model.ChannelTypePrivate:
+		if !*config.MoveThreadFromPrivateChannelEnable {
+			return errors.New("Wrangler is currently configured to not allow moving posts from private channels")
+		}
+	case model.ChannelTypeDirect:
+		if !*config.MoveThreadFromDirectMessageChannelEnable {
+			return errors.New("Wrangler is currently configured to not allow moving posts from direct message channels")
+		}
+	case model.ChannelTypeGroup:
+		if !*config.MoveThreadFromGroupMessageChannelEnable {
+			return errors.New("Wrangler is currently configured to not allow moving posts from group message channels")
+		}
+	}
+
+	if !originalChannel.IsGroupOrDirect() && !targetChannel.IsGroupOrDirect() {
+		// DM and GM channels are "teamless" so it doesn't make sense to check
+		// the MoveThreadToAnotherTeamEnable config when dealing with those.
+		if !*config.MoveThreadToAnotherTeamEnable && targetChannel.TeamId != originalChannel.TeamId {
+			return errors.New("Wrangler is currently configured to not allow moving messages to different teams")
+		}
+	}
+
+	if *config.MoveThreadMaxCount != int64(0) && *config.MoveThreadMaxCount < int64(wpl.NumPosts()) {
+		return fmt.Errorf("the thread is %d posts long, but this command is configured to only move threads of up to %d posts", wpl.NumPosts(), *config.MoveThreadMaxCount)
+	}
+
+	_, appErr := a.GetChannelMember(c, targetChannel.Id, user.Id)
```

Comment:
As mentioned above there are definitely some intricacies that we'll need to address before this could be fully secure - what you've mentioned is definitely possible so would be on that list.